pf/tfbridge/ignore_changes_test.go

```diff
@@ -17,205 +17,132 @@ package tfbridge
+			path:          "mapProp.foo",
 		},
 		{
-			notes:           "known map element is not ignored",
-			ignoreChanges:   []string{"mapProp.bar"},
-			path:            tftypes.NewAttributePath().WithAttributeName("map_prop").WithElementKeyString("foo"),
-			shouldNotIgnore: true,
+			notes:         "known map element is not ignored",
+			ignoreChanges: []string{"mapProp.bar"},
+			path:          "mapProp.foo",
+			expectChange:  true,
 		},
 		{
 			notes:         "any map element is ignored",
 			ignoreChanges: []string{"mapProp[*]"},
-			path:          tftypes.NewAttributePath().WithAttributeName("map_prop").WithElementKeyString("foo"),
-		},
-		{
-			notes:           "output-only properties are not ignored",
-			ignoreChanges:   []string{"outProp"},
-			path:            tftypes.NewAttributePath().WithAttributeName("out_prop"),
-			shouldNotIgnore: true,
+			path:          "mapProp.foo",
 		},
 		{
-			notes:         "named object properties are ignored",
-			ignoreChanges: []string{"refProp.objProp"},
-			path:          tftypes.NewAttributePath().WithAttributeName("ref_prop").WithAttributeName("obj_prop"),
+			notes:         "new elements are removed if ignored",
+			ignoreChanges: []string{"new"},
+			path:          "new",
 		},
 		{
-			notes:         "resource refs can be ignored (without recurring into the properties)",
-			ignoreChanges: []string{"resourceProp"},
-			path:          tftypes.NewAttributePath().WithAttributeName("resourceProp"),
-		},
-		{
-			notes:         "flattened maxitems=1 list paths can be ignored",
-			ignoreChanges: []string{"flatListProp.objProp"},
-			path: tftypes.NewAttributePath().WithAttributeName("flat_list_prop").
-				WithElementKeyInt(1).WithAttributeName("obj_prop"),
+			notes:         "old elements are preserved if ignored",
+			ignoreChanges: []string{"old"},
+			path:          "old",
 		},
 	}
 
 	for _, c := range cases {
-		ic, err := newIgnoreChanges(schema, tokens.Token(token), &testRenames{}, c.ignoreChanges)
-		require.NoError(t, err)
-		actual := ic.IsIgnored(c.path)
-		assert.Equalf(t, !c.shouldNotIgnore, actual, c.notes)
-	}
-}
+		c := c
+		t.Run(c.path+":"+c.notes, func(t *testing.T) {
+			old, new := old(), new()
+			path, err := resource.ParsePropertyPath(c.path)
+			require.NoError(t, err)
 
-type ignoreChangesTestCase struct {
-	ignoreChanges   []string
-	path            *tftypes.AttributePath
-	shouldNotIgnore bool
-	notes           string
-}
+			new, err = applyIgnoreChanges(old, new, c.ignoreChanges)
+			require.NoError(t, err)
 
-// Use default name manglers.
-type testRenames struct{}
+			prev, _ := path.Get(resource.NewObjectProperty(old))
+			value, _ := path.Get(resource.NewObjectProperty(new))
 
-func (*testRenames) PropertyKey(_ tokens.Token,
-	property convert.TerraformPropertyName, _ tftypes.Type) resource.PropertyKey {
-	return resource.PropertyKey(tfbridge.TerraformToPulumiNameV2(property, nil, nil))
-}
+			if c.expectChange {
+				assert.NotEqual(t, prev, value)
```

Comment:
Is assert.Equal same as prev.DeepEqual(value)?

Reply:
It has better error messages, but it's a deep equals if that's what you're asking. It is at least as semantically rigorous as `prev.DeepEquals`.